configure.ac

```
@@ -19,6 +19,16 @@ if test "$PKG_CONFIG" = ""; then
   AC_MSG_ERROR([pkg-config not found])
 fi
 
+if test -n "$PKG_CONFIG_PATH"; then
+  PKG_CONFIG="PKG_CONFIG_PATH=$PKG_CONFIG_PATH $PKG_CONFIG"
+fi
+if test -n "$PKG_CONFIG_LIBDIR"; then
+  PKG_CONFIG="PKG_CONFIG_LIBDIR=$PKG_CONFIG_LIBDIR $PKG_CONFIG"
+fi
+if test -n "$PKG_CONFIG_PATH" || test -n "$PKG_CONFIG_LIBDIR"; then
+  PKG_CONFIG="env $PKG_CONFIG"
+fi
```

nit: Maybe this is nicer (shorter, less coupled):

```suggestion
if test -n "$PKG_CONFIG_PATH"; then
  PKG_CONFIG="env PKG_CONFIG_PATH=$PKG_CONFIG_PATH $PKG_CONFIG"
fi
if test -n "$PKG_CONFIG_LIBDIR"; then
  PKG_CONFIG="env PKG_CONFIG_LIBDIR=$PKG_CONFIG_LIBDIR $PKG_CONFIG"
fi
```

This may create two `env` invocations but it still works.

---

one-liner, not sure if it's better ;)
```
PKG_CONFIG="env ${PKG_CONFIG_PATH+PKG_CONFIG_PATH=}${PKG_CONFIG_PATH} ${PKG_CONFIG_LIBDIR+PKG_CONFIG_LIBDIR=}${PKG_CONFIG_LIBDIR} $PKG_CONFIG"
```

---

I don't really mind which of the 3 versions we pick. The shorter less-coupled version seems fine.

---

Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/25687#issuecomment-1196481084).